panic: Fts5PhraseIter redefined

the error log file of running 'llcppgtest sqlite3' command :

```
2024/12/18 13:21:43 Fts5PhraseIter redefined
panic: Fts5PhraseIter redefined

goroutine 1 [running]:
log.Panicln({0xc00039d858?, 0xe?, 0xc000461e60?})
	/usr/local/Cellar/go@1.21/1.21.13/libexec/src/log/log.go:446 +0x5a
github.com/goplus/llcppg/cmd/gogensig/convert.substObj(0xc00007e360, 0x0?, {0xc0007e71a0, 0xe}, {0x13ae878?, 0xc000a4fd10})
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/convert/type.go:417 +0x1d9
github.com/goplus/llcppg/cmd/gogensig/convert.(*Package).NewTypeDecl(0xc00067eac0, 0xc000a84420)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/convert/package.go:248 +0x465
github.com/goplus/llcppg/cmd/gogensig/convert.(*AstConvert).VisitStruct(0x18?, 0xc00039da00?, 0x2?, 0xc000a84420)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/convert/convert.go:99 +0x2e
github.com/goplus/llcppg/cmd/gogensig/visitor.(*BaseDocVisitor).visitStruct(...)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/visitor/ast_visitor.go:88
github.com/goplus/llcppg/cmd/gogensig/visitor.(*BaseDocVisitor).visitTypeDecl(0xc000a60520?, 0xc0006be180?)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/visitor/ast_visitor.go:73 +0x71
github.com/goplus/llcppg/cmd/gogensig/visitor.(*BaseDocVisitor).visitNode(0xc000a05b60?, {0x12cab40?, 0xc000a84420?})
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/visitor/ast_visitor.go:35 +0x54
github.com/goplus/llcppg/cmd/gogensig/visitor.(*BaseDocVisitor).Visit(0xc000a05b58?, {0x12b5de0?, 0xc000654320?})
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/visitor/ast_visitor.go:49 +0x6b
github.com/goplus/llcppg/cmd/gogensig/processor.(*DocVisitorManager).Visit(0xc000061200?, {0x12b5de0, 0xc000654320}, {0xc00050e6f0, 0x27}, {0xc0000c69d3, 0x9}, 0x1?)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/processor/processor.go:23 +0xa6
github.com/goplus/llcppg/cmd/gogensig/convert/basic.ConvertProcesser.func1(0x12cd900?)
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/convert/basic/basic.go:38 +0x39
github.com/goplus/llcppg/cmd/gogensig/processor.(*DocFileSetProcessor).visitFile(0xc000578200, {0xc00050e6f0, 0x27}, {0xc00007b200?, 0x3, 0x4})
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/processor/processor.go:77 +0x2ce
github.com/goplus/llcppg/cmd/gogensig/processor.(*DocFileSetProcessor).ProcessFileSet(0xc000578200, {0xc00007b200?, 0x3, 0x4})
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/processor/processor.go:95 +0x17d
main.main()
	/Users/xulingjiao/Documents/GitHub/tsingbx-llcppg/cmd/gogensig/gogensig.go:94 +0x4cf
panic: exit status 2
```

[err.log](https://github.com/user-attachments/files/18175440/err.log)
and the llcppg.cfg:
```json
{
	"name": "sqlite3",
	"cflags": "$(pkg-config --cflags sqlite3)",
	"libs": "$(pkg-config --libs sqlite3)",
	"include": [
		"sqlite3.h",
		"sqlite3ext.h"
	],
	"deps": null,
	"trimPrefixes": [],
	"cplusplus": false
}
```

--- a/cmd/gogensig/convert/_testdata/forwarddecl/conf/llcppg.symb.json
+++ b/cmd/gogensig/convert/_testdata/forwarddecl/conf/llcppg.symb.json
@@ -1,8 +1,12 @@
 [
-    {
-      "mangle": "lua_getstack",
-      "c++": "lua_getstack",
-      "go": "Getstack"
-    }
-  ]
-  
+  {
+    "mangle": "lua_getstack",
+    "c++": "lua_getstack",
+    "go": "Getstack"
+  },
+  {
+    "mangle": "fts5_extension_function",
+    "c++": "fts5_extension_function",
+    "go": "Fts5ExtensionFunction"
+  }
+]

--- a/cmd/gogensig/convert/_testdata/forwarddecl/gogensig.expect
+++ b/cmd/gogensig/convert/_testdata/forwarddecl/gogensig.expect
@@ -84,12 +84,41 @@ type CallInfo struct {
 	Unused [8]uint8
 }
 
+type Fts5ExtensionApi struct {
+	Unused [8]uint8
+}
+
+type Fts5Context struct {
+	Unused [8]uint8
+}
+
+type Fts5PhraseIter struct {
+	A *int8
+	B *int8
+}
+
+type Value struct {
+	Unused [8]uint8
+}
+
+type Context struct {
+	Unused [8]uint8
+}
+// llgo:type C
+type Fts5ExtensionFunction func(*Fts5ExtensionApi, *Fts5Context, *Context, c.Int, **Value)
+
 ===== llcppg.pub =====
+Fts5Context
+Fts5ExtensionApi
+Fts5PhraseIter
 bar Bar
+fts5_extension_function Fts5ExtensionFunction
 lua_Debug Debug
 lua_State State
+sqlite3_context Context
 sqlite3_file File
 sqlite3_io_methods IoMethods
 sqlite3_pcache Pcache
 sqlite3_pcache_methods2 PcacheMethods2
 sqlite3_pcache_page PcachePage
+sqlite3_value Value

--- a/cmd/gogensig/convert/_testdata/forwarddecl/hfile/temp.h
+++ b/cmd/gogensig/convert/_testdata/forwarddecl/hfile/temp.h
@@ -5,7 +5,6 @@ struct bar
     foo *a;
 };
 
-
 typedef struct sqlite3_file sqlite3_file;
 struct sqlite3_file
 {
@@ -72,4 +71,23 @@ struct lua_Debug
     char short_src[LUA_IDSIZE];
     /* private part */
     struct CallInfo *i_ci; /* active function */
+};
+
+typedef struct Fts5ExtensionApi Fts5ExtensionApi;
+typedef struct Fts5Context Fts5Context;
+typedef struct Fts5PhraseIter Fts5PhraseIter;
+
+typedef struct sqlite3_value sqlite3_value;
+typedef struct sqlite3_context sqlite3_context;
+typedef void (*fts5_extension_function)(const Fts5ExtensionApi *pApi, /* API offered by current FTS version */
+                                        Fts5Context *pFts,            /* First arg to pass to pApi functions */
+                                        sqlite3_context *pCtx,        /* Context for returning result/error */
+                                        int nVal,                     /* Number of values in apVal[] array */
+                                        sqlite3_value **apVal         /* Array of trailing arguments */
+);
+
+struct Fts5PhraseIter
+{
+    const unsigned char *a;
+    const unsigned char *b;
 };

--- a/cmd/gogensig/convert/package.go
+++ b/cmd/gogensig/convert/package.go
@@ -553,18 +553,19 @@ func (p *Package) DeclName(name string) (pubName string, changed bool, err error
 
 // For a decl name, if it's a current package, remove the prefixed name
 func (p *Package) GetGoName(name string) (pubName string, changed bool) {
-	prefixes := []string{}
+	if goName, exists := p.Pubs[name]; exists {
+		if goName == "" {
+			return name, false
+		}
+		return goName, goName != name
+	}
+	var prefixes []string
 	if p.curFile.inCurPkg {
 		prefixes = p.CppgConf.TrimPrefixes
 	}
-	goName, ok := p.Pubs[name]
-	if ok {
-		pubName = goName
-	} else {
-		pubName = names.GoName(name, prefixes)
-	}
-	changed = pubName != name
-	return
+
+	pubName = names.GoName(name, prefixes)
+	return pubName, pubName != name
 }
 
 func (p *Package) CollectNameMapping(originName, newName string) {

